src/database/mod.rs

```diff
@@ -891,21 +892,32 @@ mod test {
         chrono::Duration::milliseconds(x.num_milliseconds().abs())
     }
 
-    // TODO: we should probably consolidate all tests that propagate errors to
-    // TODO: either use anyhow or eyre
-    async fn setup_db() -> anyhow::Result<(Database, DockerContainerGuard)> {
-        let db_container = postgres_docker_utils::setup().await?;
-        let db_socket_addr = db_container.address();
-        let url = format!("postgres://postgres:postgres@{db_socket_addr}/database");
+    struct DBSetuper {
+        docker: Cli,
+    }
 
-        let db = Database::new(&DatabaseConfig {
-            database:        SecretUrl::from_str(&url)?,
-            migrate:         true,
-            max_connections: 1,
-        })
-        .await?;
+    fn get_db_setuper() -> DBSetuper {
+        DBSetuper {
+            docker: Cli::default(),
+        }
+    }
 
-        Ok((db, db_container))
+    impl DBSetuper {
+        // TODO: we should probably consolidate all tests that propagate errors to
+        // TODO: either use anyhow or eyre
+        async fn setup_db<'a>(&'a self) -> anyhow::Result<(Database, DockerContainer<'a>)> {
```

Comment:
Can we ditch `DBSetuper` and make it into 
```
async fn setup_db() -> anyhow::Result<(Database, DockerContainer<'a>)> {
    let docker = Cli::default();
    // ...
}
```

Reply:
As discussed - refactored.